### ISSUE TITLE
Cannot start at OffsetEnd and OffsetStored on two different topics

### ISSUE THREAD

**Issue by rikardev** [confluentinc/confluent-kafka-go#580]
Description
===========
When subscribing on two topics, cannot start/assign at OffsetEnd on one and OffsetStored on the other. OffsetEnd and OffsetBeginning seems to work. Looking at the code I think this goes down to the C implementation?

How to reproduce
================
Noticed this in our production applications but have reduced our code to a minimal example on which the issue is reproducible with a local instance of kafka. I have two topics each with 50 partitions onto which I have produced 1 million messages each using the following snippet:

```
package main

import (
	"fmt"
	"github.com/confluentinc/confluent-kafka-go/kafka"
	"log"
	"os"
	"os/signal"
	"syscall"
	"time"
)

func main() {
	sigchan := make(chan os.Signal, 1)
	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)

	var topic1 = "test-seek-end-1"
	var topic2 = "test-seek-end-2"

	var logger = log.New(os.Stdout, "", 0)

	cm := kafka.ConfigMap{
		"bootstrap.servers":           "localhost:9092",
		"go.produce.channel.size":     16384,
		"partitioner":                 "murmur2_random",
		"queue.buffering.max.ms":      10,
		"statistics.interval.ms":      30000,
		"socket.keepalive.enable":     true,
		"socket.send.buffer.bytes":    262144,
		"socket.receive.buffer.bytes": 32768,
		"go.batch.producer":           true,
		//"acks":                   1,
	}

	kafkaProducer, err := kafka.NewProducer(&cm)
	if err != nil {
		panic(err)
	}

	toProduce := int64(1)
	// added a time for each message to be debug using offset for timestamp
	tsUnixStart := time.Now().Unix() - toProduce

	go func() {
		for e := range kafkaProducer.Events() {
			switch ev := e.(type) {
			case *kafka.Message:
				m := ev
				if m.TopicPartition.Error != nil {
					logger.Printf("Delivery failed: %v\n", m.TopicPartition.Error)
				}
			default:
				logger.Printf("Ignored event: %T\n", ev)
			}
		}
	}()


	for _, topic := range []string{topic1, topic2} {
		for i := int64(0); i < toProduce; i++ {

			kafkaProducer.ProduceChannel() <- &kafka.Message{
				TopicPartition: kafka.TopicPartition{
					Topic:     &topic,
					Partition: kafka.PartitionAny,
				},
				Value:     []byte(fmt.Sprintf("%d", i)),
				Key:       []byte(fmt.Sprintf("%d", i)),
				Timestamp: time.Unix(tsUnixStart + i, 0),
			}

			if i%1e5 == 0 {
				logger.Printf("Produced another batch for %s, total: %d\n", topic, i)
			}
		}
	}

	logger.Println("Done producing, waiting for close")

	kafkaProducer.Flush(60_000)
	kafkaProducer.Close()

	logger.Println("Done producing")

}
```

Then I start the consumer with the following snippet. Here I when I assign I say that `test-seek-end-1` should start at the end and `test-seek-end-2`:

```
package main

import (
	"github.com/confluentinc/confluent-kafka-go/kafka"
	"log"
	"os"
	"os/signal"
	"syscall"
)

func main() {
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)

	var topic1 = "test-seek-end-1"
	var topic2 = "test-seek-end-2"
	var topics = []string {
		topic1,
		topic2,
	}

	var logger = log.New(os.Stdout, "", 0)

	cm := kafka.ConfigMap{
		"bootstrap.servers":               "localhost:9092",
		"group.id":                        "kafka-offset-debug",
		"enable.partition.eof":            true,
		"go.events.channel.enable":        false,
		"go.application.rebalance.enable": true,
		"default.topic.config":            kafka.ConfigMap{"auto.offset.reset": "earliest"},
		"statistics.interval.ms":          30000,
		"isolation.level":                 "read_uncommitted",
		"socket.keepalive.enable":         true,
		"socket.send.buffer.bytes":        262144,
		"socket.receive.buffer.bytes":     32768,
		"queued.min.messages":             1000,
		"session.timeout.ms":              20000,
	}

	kafkaConsumer, err := kafka.NewConsumer(&cm)

	if err != nil {
		logger.Panicf("Failed to create Consumer: %s\n", err)
	}

	err = kafkaConsumer.SubscribeTopics(topics, nil)
	if err != nil {
		logger.Panicf("Failed to subscribe: %s\n", err)
	}

EventLoop:
	for {
		select {
		case sig := <-sigChan:
			logger.Printf("Caught signal %v: terminating\n", sig)
			break EventLoop
		default:
			ev := kafkaConsumer.Poll(1000)
			if ev == nil {
				logger.Println("Got no events on poll")
			}

			switch e := ev.(type) {
			case kafka.AssignedPartitions:
				logger.Println("Got assign")
				assignPartitions := make([]kafka.TopicPartition, len(e.Partitions))
				for i, tp := range e.Partitions {
					assignPartitions[i] = tp
					if *tp.Topic == topic1 {
						assignPartitions[i].Offset = kafka.OffsetEnd
					} else if *tp.Topic == topic2 {
						assignPartitions[i].Offset = kafka.OffsetStored
					} else {
						logger.Printf("Found unexpeted: %v", tp)
					}
				}

				err = kafkaConsumer.Assign(assignPartitions)
				if err != nil {
					logger.Panicf("Failed to assign partitions %v: %v", assignPartitions, err)
				}

			case kafka.RevokedPartitions:
				logger.Println("Got revoke")
				err := kafkaConsumer.Unassign()
				if err != nil {
					logger.Panicf("Failed to revoke partitions %v: %v", e.Partitions, err)
				}

			case *kafka.Message:
				logger.Printf(
					"Message: %s, %d, %s",
					*e.TopicPartition.Topic, e.TopicPartition.Partition, e.TopicPartition.Offset,
				)
				//time.Sleep(10 * time.Millisecond)
			}
		}
	}
}
```

**Expected behavior:** "test-seek-end-1" starts consuming from the end (i.e I should not receive any messages) and "test-seek-end-2" to start from the beginning (or where it last read if it isn't the first time I run the code).

**Actual behaviour:** "test-seek-end-1" starts consuming from the beginning

Checklist
=========
Please provide the following information:

 - [X] confluent-kafka-go and librdkafka version (`LibraryVersion()`): confluent-kafka-go: v1.5.2 with bundled release of librdkafka: 1.5.2
 - [X] Apache Kafka broker version: 2.3.0
 - [X] Client configuration: See consumer snippet above
 - [X] Operating system: MacOS local but also tested with both Alpine and Archlinux in production.
 - [ ] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [X] Critical issue: I would argue yes but not sure if this is for me to determine

**Comment by rikardev** [comment]
I realize I should clarify what happens here so I've edited the original report to include this:

**Expected behavior:** "test-seek-end-1" starts consuming from the end (i.e I should not receive any messages) and "test-seek-end-2" to start from the beginning (or where it last read if it isn't the first time I run the code).

**Actual behaviour:** "test-seek-end-1" starts consuming from the beginning

**Comment by edenhill** [comment]
Yep, this is a bug in librdkafka which is fixed in the upcoming v1.6.0 release

**Comment by rikardev** [comment]
That's great, thank you!

Spent an evening digging through the code in librdkafka and couldn't make sense of why it didn't work. Guess I shouldn't have been on master branch to begin with.
